draft-stark-expect-ct.md

```diff
@@ -23,6 +23,316 @@ TODO: Wite an abstract here
+domain name and its associated Expect-CT directives in non-volatile storage. The
+domain name and associated Expect-CT directives are collectively known as
+"Expect-CT metadata".
+
+The UA MUST note a host as an Expect-CT host if and only if it received the
+Expect-CT response header field over an error-free TLS connection, inluding the
+validation added in {{expect-ct-compliance}}, that included the `enforce`
+directive.
+
+To note a host as an Expect-CT host, the UA MUST set its Expect-CT metadata to
+the effectie expiration date and report-uri (if any) given in the most recently
+received valid Expect-CT header.
+
+For forward compatibility, the UA MUST ignore any unrecognized Expect-CT header
+directives, while still processing those directives it does
+recognize. {{response-header-field-syntax} specifies the directives `enforce`,
+`max-age`, and `report-uri`, but future specifications and implementations might
+use additional directives.
+
+## Evaluating Expect-CT Connections for CT Compliance {#expect-ct-compliance}
+
+When a UA connects to an Expect-CT host using a TLS connection, if the TLS
+connection has errors, the UA MUST terminate the connection without allowing the
+user to proceed anyway. (This behavior is the same as that required by
+{{!RFC6797}}.)
+
+If the connection has no errors, then the UA will apply an additional
+correctness check: compliance with a CT policy. A UA should evaluate compliance
+with its CT policy whenever connecting to an Expect-CT host, as soon as
+possible. It is acceptable to skip this CT compliance check for some hosts
+according to local policy. For example, a UA may disable CT compliance checks
+for hosts whose validated certificate chain terminates at a user-defined trust
+anchor, rather than a trust anchor built-in to the UA (or underlying platform).
+
+SOMETHING SOMETHING about the UA getting to choose the policy based on the
+SCTs. Example policies in an appendix? Something advising UAs to publish their
+policies somewhere? Something about using reporting to discover policy changes?
+
+A UA that has previously noted a host as an Expect-CT host MUST evaluate
+evaluate CT compliance when setting up the TLS session, before beginning an HTTP
+conversation over the TLS channel.
+
+If the UA does not evaluate CT compliance, e.g. because the user has elected to
+disable it, or because a presented certificate chain chains up to a user-defined
+trust anchor, UAs SHOULD NOT send Expect-CT reports.
+
+# Interactions with Preloaded Pin Lists
+
+# Reporting Expect-CT Failure
+
+When the UA receives an Expect-CT header with a `report-uri` directive that does
+not comply with the UA's CT policy, or when the UA connects to a noted Expect-CT
+host that does not comply with the CT policy, the UA SHOULD report Expect-CT
+failures to the configured `report-uri`. The UA does this by POSTing a JSON
+{{!RFC7159}} message to the URI. The JSON message takes this form:
```

Comment:
Recommend something like this... `webappapis#*` pointing to http://www.w3.org/TR/html5/webappapis.html and `infrastructure#fetch` to http://www.w3.org/TR/html5/infrastructure.html#fetch

> 1. Prepare a JSON object report object with a single key, `expect-ct-report`, whose value is the result of generating a violation report object. 
2. Let `report body` be the JSON stringification of report object. 
3. {{webappapis#Queue a task}} to {{infrastructure#fetch}} `report-uri` from the origin of the protected resource, with the synchronous flag *not* set, using HTTP method `POST`, with a `Content-Type` header field of `application/expect-ct-report`, and an entity body consisting of `report body`. If the origin of `report URL` is not the same as the origin of the protected resource, the block cookies flag MUST also be set. The user agent MUST NOT follow redirects when fetching this resource. (Note: The user agent ignores the fetched resource.) The {{webappapis#task source}} for these {{webappapis#concept-tasks}} is the **Content Security Policy task source**.